src/components/OrderDetailWidget/OrderDetailWidget.tsx

```diff
@@ -162,6 +170,26 @@ const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ order }) => {
     );
   };
 
+  useEffect(() => {
+    if (!indexerUrls && library) {
+      dispatch(fetchIndexerUrls({ provider: library }));
+    }
+  }, [dispatch, indexerUrls, library]);
+
+  useEffect(() => {
+    if (indexerUrls) {
+      dispatch(
+        getFilteredOrders({
+          filter: {
+            senderTokens: [senderToken?.address!],
+            signerTokens: [signerToken?.address!],
+            page: 1,
+          },
+        })
+      );
+    }
+  }, [dispatch, indexerUrls, senderToken, signerToken]);
```

Comment:
```suggestion
  useEffect(() => {
    if (indexerUrls && senderToken && signerToken) {
      dispatch(
        getFilteredOrders({
          filter: {
            senderTokens: [senderToken.address, signerToken.address],
            signerTokens: [signerToken.address, senderToken.address],
            page: 1,
          },
        })
      );
    }
  }, [indexerUrls, senderToken, signerToken]);
```

You want the order both ways!

Reply:
Not too sure we do? The taker is probably only liquid in one direction.

Reply:
Oh yeah ofcourse the taker only wants one direction.